vms/evm/subscriber.go

```diff
@@ -154,25 +159,33 @@ func (s *subscriber) ProcessFromHeight(height *big.Int) error {
 		return err
 	}
 
-	// Cap the number of blocks to process to MaxBlocksToProcess
-	toBlock := big.NewInt(0).SetUint64(latestBlock)
-	if height.Cmp(big.NewInt(0).Sub(toBlock, big.NewInt(MaxBlocksToProcess))) < 0 {
-		s.logger.Warn(
-			fmt.Sprintf("Requested to process too many blocks. Processing only the most recent %d blocks", MaxBlocksToProcess),
-			zap.String("requestedBlockHeight", height.String()),
-			zap.String("latestBlockHeight", toBlock.String()),
-			zap.String("chainID", s.chainID.String()),
-		)
-		height = big.NewInt(0).Add(toBlock, big.NewInt(-MaxBlocksToProcess))
+	bigLatestBlock := big.NewInt(0).SetUint64(latestBlock)
+
+	for fromBlock := big.NewInt(0).Set(height); fromBlock.Cmp(bigLatestBlock) <= 0; fromBlock.Add(fromBlock, big.NewInt(MaxBlocksPerRequest+1)) {
+		toBlock := big.NewInt(0).Add(fromBlock, big.NewInt(MaxBlocksPerRequest))
```

Comment:
I think there's an off-by-one-error here. Say we have `MaxBlocksPerRequest = 20`, and `fromBlock = 0`. Then `toBlock` would equal `20`, and since `eth_getLogs` takes an inclusive block range, we'd process `21` blocks instead of `20`. The fix here is to use `MaxBlocksPerRequest-1` on this line, and `MaxBlocksPerRequest` in the loop condition.

Reply:
(This is assuming `eth_getLogs` does in fact take an inclusive block range)

Reply:
`eth_getLogs` is inclusive on both side, great call out. I don't think it makes a _huge_ difference either way, but it is odd if `MaxBlockPerRequest` isn't the true effective value.

Reply:
should we set `toBlock := fromBlock + maxBlockPerRequest - 1`? Since right now I think we'd be processing maxBlockPerRequest + 1

Reply:
Thanks guys, really appreciate the scrutiny here. Addressed in https://github.com/ava-labs/awm-relayer/pull/89/commits/a59dd14cecbf9e05463522ceb1b775c537fc0128